person.rb

```diff
@@ -0,0 +1,21 @@
+class Person
+  attr_accessor :name, :age
+  attr_reader :id
+
+  def initialize(age, name: 'Unknown', parent_permission: true)
+    @id = nil
+    @name = name
+    @age = age
+    @parent_permission = parent_permission
+  end
+
+  def can_use_services?
+    true if of_age? || @parent_permission
+  end
+
+  private
+
+  def of_age?
+    true if age >= 18
+  end
+end
```

Comment:
- [ ] [OPTIONAL]   you don’t need to write true explicitly in Ruby, as the if condition will return true or false by itself. Here is a possible way to improve your code:

```rb
def can_use_services?
  of_age? || @parent_permission
end

private

def of_age?
  age >= 18
end

```

Reply:
Alright! thank you so much for these corrections.